### PR TITLE
Multistep cover learned equivalent ground truth

### PR DESCRIPTION
**Pull request by ashayathalye**
This PR adds two more options to cover_multistep_options: LearnedEquivalentPick and LearnedEquivalentPlace. These options use the same parameterization that the options learned through _SimpleOptionLearner use. Corresponding "ground truth" samplers for these two options are added to oracle_approach. A few minor changes are made to update the control flow for how the cover NSRTs are defined in oracle_approach, as well as a few runtime bug fixes such as the cover_multistep_option non-learned-equivalent samplers not properly doing relative positioning (it was over-constraining the support of the uniform distribution).

### CODE REVIEW

**Review by tomsilver — COMMENTED** [review]
@ashayathalye looking very good, thank you! let's do what you mentioned on slack as part of this PR -- refactor the oracle covers stuff to improve readability.

**Comment by ronuchit** [comment]
I'll review after @tomsilver 's suggested refactor!

File: src/envs/cover.py
Comment (ashayathalye) [reply src/envs/cover.py:658]: If the grip value is greater than 0, the gripper is in the ON position. If it is less than or equal to 0, the gripper is in the OFF position. I thought it's less confusing if the value we set is not the threshold value, since someone may not remember if the logic is is > or >= to set it to ON. 

I now changed it to +1.0 and -1.0 instead of +/- 0.1, as 0.1 is more arbitrary than +/- 1.0

File: src/approaches/oracle_approach.py
Comment (ashayathalye) [reply src/approaches/oracle_approach.py:241]: Good point, I added checks as you suggested.

File: src/approaches/oracle_approach.py
Comment (ronuchit) [inline src/approaches/oracle_approach.py:91]: Thank you for refactoring this! It had been annoying to me how complicated it was getting. Looks really clean now!!

File: src/approaches/oracle_approach.py
Comment (ronuchit) [inline src/approaches/oracle_approach.py:166]: A bit confusing, would prefer it written out normally:
```
if CFG.env == "cover_multistep_options":
    assert len(objs) == 2
else:
    assert len(objs) == 1
```

File: src/approaches/oracle_approach.py
Comment (ronuchit) [inline src/approaches/oracle_approach.py:238]: A bit confusing, would prefer it written out normally:
```
if CFG.env == "cover_multistep_options":
    assert len(objs) == 3
else:
    assert len(objs) == 2
```

File: tests/envs/test_cover.py
Comment (ronuchit) [inline tests/envs/test_cover.py:176]: indent this line to align with `Pick` in the previous line

**Review by ronuchit — APPROVED** [review]
LGTM! just a few minor comments. but since i generally don't know what's going on, please wait for @tomsilver's approval before merging :)

File: src/approaches/oracle_approach.py
Comment (tomsilver) [inline src/approaches/oracle_approach.py:137]: minor, can you reorder these so that we don't end up creating and then immediately overriding `option` and `option_vars`? one possibility is to move this case up before the previous one, so:

```
if CFG.env in ("cover", "cover_hierarchical_types"):
...
elif CFG.env == "cover_multistep_options" and CFG.cover_multistep_use_learned_equivalents:
...
else:
    assert CFG.env in ("cover_typed_options", "cover_multistep_options")
```

this comment also applies to a similar pattern with Place

**Review by tomsilver — APPROVED** [review]
excellent work! we can merge after addressing the few minor points

File: tests/approaches/test_nsrt_learning_approach.py
Comment (ronuchit) [inline tests/approaches/test_nsrt_learning_approach.py:68]: After you rebase to most recent master, I'd recommend also adding `sampler_learner="random"` in addition to `try_solving=False` for extra speed.

**Review by ronuchit — APPROVED** [review]
LGTM

**Pull request merged**
